python/revng/daemon/__init__.py

```diff
@@ -131,6 +132,7 @@ def generate_context(request: Request):
         return {
             "manager": manager,
             "event_manager": event_manager,
+            "index_lock": asyncio.Lock(),
```

Comment:
Why do we need a lock?
Docs.

Reply:
Without the lock, there's a TOCTOU between the check of the index and the actual analysis is being run
ACK on adding docs

Reply:
Added docs